common.h

```diff
@@ -32,7 +32,24 @@ static inline uint64_t mmio_read64(void *addr)
 	low = le32_to_cpu(*p);
 	high = le32_to_cpu(*(p + 1));
 
-	return ((uint64_t) high << 32) | low;
+	return ((uint64_t)high << 32) | low;
 }
 
+static inline void mmio_write32(void *addr, uint32_t value)
+{
+	leint32_t *p = addr;
+
+	*p = cpu_to_le32(value);
+}
+
+/* Access 64-bit registers as 2 32-bit; Some devices fail 64-bit MMIO. */
+static inline void mmio_write64(void *addr, uint64_t value)
```

Comment:
This sounds like a very dangerous thing to do. Wouldn't it better to use 64 bit writes per default and have a flag to fallback to 32 bit writes. These commands are very dangerous. How does the kernel figure out that things have changed anyway?